src/interchange/src/avro.rs

```diff
@@ -1395,6 +1435,29 @@ impl fmt::Debug for Decoder {
     }
 }
 
+/// Manages decoding of Avro-encoded bytes.
+pub struct Decoder {
+    csr_avro: ConfluentAvroResolver,
+    envelope: EnvelopeType,
+    debezium_dedup: Option<DebeziumDeduplicationState>,
+    debug_name: String,
+    worker_index: usize,
+    buf1: Vec<u8>,
+    buf2: Vec<u8>,
+    packer: RowPacker,
+}
+
+impl fmt::Debug for Decoder {
```

Comment:
I know this is marked as a todo but you can derive `Debug` for `Decoder` now - might be worth it to do so as part of this change?